app/models/user.rb

```diff
@@ -293,10 +293,34 @@ def teacher?
     given_lectures.any?
   end
 
+  def teachable_editor?
+    edited_courses.any? || edited_lectures.any?
+  end
+
+  def teachable_editor_or_teacher?
+    teachable_editor? || teacher?
+  end
+
+  def can_edit_teachables?
+    admin? || teachable_editor_or_teacher?
+  end
+
+  # if you are not a teacher of lecture or a module editor,
+  # but just an additional editor of some lecture, you
+  # will not be considered active if all of your edited lectures
+  # are older than one year
+  def active_teachable_editor?
+    return false unless can_edit_teachables?
+    return true if admin || course_editor? || teacher?
+
+    edited_lectures.select { |l| l.term.nil? || !stale? }
+                   .any?
+  end
+
   # a user is an editor iff he/she is a course editor or lecture editor or
   # editor of media that are not associated to talks
```

Comment:
Comment should not specify what a teachable_editor is. Instead specify it as comment for the `teachable_editor` method.

Reply:
I do not quite understand the comment (but I admit that sometimes even I get confused about the different sorts of editors, hence the comment). The comment describes what an editor is, not what a teachable_editor is: You can be an editor (e.g. if you have editing rights to some medium that is not associated to a talk in a seminar) without being a teachable_editor.

Reply:
True, but right now you also include the definition of a `teachable_editor` in the comment for `editor?` by saying an user is an editor "iff he/she is a course editor or lecture editor or ...". If we changed the definition of `teachable_editor` in the future, we would have to remember: "Oh, we use `teachable_editor` in the `editor?` method as well and must therefore also change the comment over there. I know this is nitpicking and maybe it's just fine to leave it here, but then comments might get stale themselves more quickly.

Reply:
Ah, thank you for the explanation, now I understand. Will change it.